### PR TITLE
Implement search bar component

### PR DESCRIPTION
**Pull request by irenenguyen1017**
Hi team, I've finished one of the last features of our app. Instead of providing a description here, I've commented on each line of code to explain its purpose. If you're happy with them, please click on `resolve conversation` and approve the PR. Thank you.

[Trello ticket](https://trello.com/c/6mebcfp3/92-implement-search-bar-component)

### CODE REVIEW

File: src/components/CartInHeader.jsx [hunk 1]
```diff
@@ -7,9 +7,11 @@ import { Link } from 'react-router-dom';
 export default function CartInHeader() {
   const { getCartTotalQuantity } = useContext(CartContext);
 
+  const totalItem = getCartTotalQuantity();
+
   return (
-    <Link to='/cart'>
-      <Badge color='secondary' content={getCartTotalQuantity()} isInvisible={false} shape='circle'>
+    <Link to='/cart' className='inline-flex items-center'>
+      <Badge color='secondary' content={totalItem} isInvisible={totalItem === 0} shape='circle'>
```
Comment (irenenguyen1017) [inline src/components/CartInHeader.jsx:14]: When the cart is empty, the cart icon does not display a number.

File: src/components/ProductCardList.jsx [hunk 2]
```diff
@@ -1,17 +1,33 @@
...
+import { Skeleton } from '@nextui-org/react';
+
+const ProductCardList = ({
+  products,
+  title,
+  isHorizontalViewInMobile = true,
+  isLoading,
+  skeletonNo = 5
+}) => {
+  const className = isHorizontalViewInMobile ? 'flex overflow-x-auto' : 'grid';
```
Comment (irenenguyen1017) [inline src/components/ProductCardList.jsx:11]: `isHorizontalViewInMobile` determines how the ProductCard is displayed on mobile devices. If set to true, ProductCards will be shown in a horizontal scroll view. If false, they will be arranged in a grid layout.

File: src/components/ProductCardList.jsx [hunk 3]
```diff
@@ -1,17 +1,33 @@
...
 
-const ProductCardList = ({ products, title }) => {
   return (
     <div className='flex flex-col mt-10'>
-      <h3 className='relative text-2xl lg:text-4xl font-light after:content-[""] after:w-[60px] after:h-2 after:block after:bg-secondary after:mt-2'>
-        {title}
-      </h3>
-      <div className='flex overflow-x-auto lg:overflow-x-visible lg:grid lg:grid-cols-[repeat(auto-fill,_minmax(240px,_1fr))] gap-8 p-4 no-scrollbar'>
+      {title ? (
```
Comment (irenenguyen1017) [inline src/components/ProductCardList.jsx:15]: If the title is not provided, it will not be shown.

File: src/components/ProductCardList.jsx [hunk 4]
```diff
@@ -1,17 +1,33 @@
...
+      ) : null}
+      <div
+        className={`${className} lg:grid md:grid-cols-[repeat(auto-fill,_minmax(240px,_1fr))] gap-8 no-scrollbar`}
+      >
         {/* Iterate over the products array. */}
-        {products.map((product, index) => (
-          // key={index} provides a unique key for each ProductCard component
-          <ProductCard key={index} product={product} />
-        ))}
+        {isLoading
```
Comment (irenenguyen1017) [inline src/components/ProductCardList.jsx:24]: `isLoading` determines whether to display a list of skeleton placeholders for the product cards while the app is waiting for the data to be fetched.

File: src/pages/Search.jsx [hunk 5]
```diff
@@ -49,53 +67,32 @@ export default function SearchPage() {
...
-          <Pagination
-            showControls
-            total={totalPages}
-            initialPage={currentPage}
-            onChange={handlePageChange}
+      <div className='container max-w-full mx-auto py-6'>
+        {isLoading ? null : (
+          <h3 className='text-gray-700 text-4xl font-light mb-10'>
+            {keyword ? `Products for "${keyword}"` : `Products in category "${categorySlug}"`}
+          </h3>
```
Comment (irenenguyen1017) [inline src/pages/Search.jsx:74]: Only show the message after the data has been fetched.

File: src/pages/Search.jsx [hunk 6]
```diff
@@ -49,53 +67,32 @@ export default function SearchPage() {
...
+        )}
+        {!results.length && !isLoading ? (
+          <p>No products found for '{keyword}'</p>
+        ) : (
+          <ProductsCardList
```
Comment (irenenguyen1017) [inline src/pages/Search.jsx:79]: @Mana12011207 Here, by reusing the ProductCardList component will help us comply with DRY, thereby making our code more efficient.

File: src/pages/Search.jsx [hunk 7]
```diff
@@ -49,53 +67,32 @@ export default function SearchPage() {
...
+            products={results}
+            isHorizontalViewInMobile={false}
+            isLoading={isLoading}
+            skeletonNo={8}
           />
-        </div>
+        )}
+        {totalPages > 1 ? (
```
Comment (irenenguyen1017) [inline src/pages/Search.jsx:86]: Display pagination only if the search results span two or more pages.

**Review by Mana12011207 — APPROVED** [review]

**Review by ellenpham — APPROVED** [review]

**Pull request merged**